### ISSUE TITLE
Error recovering metadata tablet

### ISSUE THREAD

**Issue by milleruntime** [apache/accumulo#1844]
I saw 2 metadata tablets fail to recover during a long running session of the Randomwalk test.  The error that was reported read:
"Error recovering tablet !0;51e7e10b1827d193;490d5011198f9047 from log files"
Here is the stacktrace from the tserver trying to load the tablet:
```
2020-12-20 00:45:53,512 [tserver.TabletServer] WARN : exception trying to assign tablet !0;51e7e10b1827d193;490d5011198f9047 hdfs://muchoshacluster/accumulo/tables/!0/t-0000yt4
java.lang.RuntimeException: Error recovering tablet !0;51e7e10b1827d193;490d5011198f9047 from log files
        at org.apache.accumulo.tserver.tablet.Tablet.<init>(Tablet.java:499)
        at org.apache.accumulo.tserver.TabletServer$AssignmentHandler.run(TabletServer.java:2413)
        at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
        at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:64)
        at org.apache.htrace.wrappers.TraceRunnable.run(TraceRunnable.java:57)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
        at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: Unable to find recovery files for extent !0;51e7e10b1827d193;490d5011198f9047 logEntry: !0;51e7e10b1827d193; hdfs://muchoshacluster/accumulo/wal/worker1+9997/e41008f5-edcf-4e18-aa
92-ba5ff021685c
        at org.apache.accumulo.tserver.TabletServer.recover(TabletServer.java:3311)
        at org.apache.accumulo.tserver.tablet.Tablet.<init>(Tablet.java:437)
        ... 8 more
```

At some point before this the Master went into SAFE_MODE to UNLOAD_METADATA_TABLETS to UNLOAD_ROOT_TABLET to STOP so it could shut down cleanly.  I am not sure why exactly it did this, memory perhaps but it repeated shutting down a few times (and crashing) after starting back up.  It was during this time that the tserver reported the above error.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 1.10.0, 1.10.1
 - Platform:  4 node Amazon EC2 Cluster created using Muchos
 - OS: CentOS 7.5.1804
4 Walkers running Randomwalk test suite:
```
  <edge id="Image.xml" weight="1"/>
  <edge id="Sequential.xml" weight="1"/>
  <edge id="MultiTable.xml" weight="1"/>
  <edge id="Shard.xml" weight="1"/>
  <edge id="Concurrent.xml" weight="1"/>
  <edge id="Conditional.xml" weight="1"/>
  <edge id="Bulk.xml" weight="1"/>
```
All 4 walkers were running the Conditional test during this time.

**Comment by milleruntime** [comment]
Accumulo was able to recovery and everything resumed functioning properly (i.e. I can scan the full metadata table, tservers all up).  So it is likely that this was a transient error.

**Comment by milleruntime** [comment]
I attempted to replicate this error in 2.1 with no success.  I will try to replicate it in 1.10.1.

**Comment by milleruntime** [comment]
I discovered that one of the Randomwalk tests in the Concurrent suite will actually shutdown the Master.  This is most likely what was causing Master to do a CLEAN_STOP, as I can't find anywhere that does this automatically in the code. Here is the test:
https://github.com/apache/accumulo/blob/f4f43febbc3e68013d8a1bcd46d8b44275e2e55e/test/src/main/java/org/apache/accumulo/test/randomwalk/concurrent/Shutdown.java#L41

**Comment by milleruntime** [comment]
After some more Randomwalk testing in just a 1.10 Uno cluster, I was able to reproduce this error for other tables but not the metadata. In all the errors, for some reason the extent in the logEntry only had the endRow and not the prevRow.  The error gets thrown when there isn't a Finished file so I am thinking that the Tablet tried to load but recovering wasn't done sorting the WAL.  I am still not sure if this is a bug or common for recovery.  If this is common during recovery, I think we could at least improve the error message.

**Comment by milleruntime** [comment]
I saw this error happen, which I think is a similar situation.  This error does provide a message about the sort not being finished.
<pre>
2021-01-11 07:51:37,858 [tserver.TabletServer] WARN : exception trying to assign tablet !0;2ac755608c39162d;26dbd6f6192bcaee hdfs://localhost:8020/accumulo/tables/!0/t-00001z4
java.lang.RuntimeException: Error recovering tablet !0;2ac755608c39162d;26dbd6f6192bcaee from log files
        at org.apache.accumulo.tserver.tablet.Tablet.<init>(Tablet.java:499)
        at org.apache.accumulo.tserver.TabletServer$AssignmentHandler.run(TabletServer.java:2413)
        at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
        at org.apache.accumulo.tserver.ActiveAssignmentRunnable.run(ActiveAssignmentRunnable.java:64)
        at org.apache.htrace.wrappers.TraceRunnable.run(TraceRunnable.java:57)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at org.apache.accumulo.fate.util.LoggingRunnable.run(LoggingRunnable.java:35)
        at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: java.io.IOException: java.io.IOException: Sort "finished" flag not found in hdfs://localhost:8020/accumulo/recovery/5bce00d4-1920-416a-8bfb-41e4cf16e9f6
        at org.apache.accumulo.tserver.log.TabletServerLogger.recover(TabletServerLogger.java:608)
        at org.apache.accumulo.tserver.TabletServer.recover(TabletServer.java:3315)
        at org.apache.accumulo.tserver.tablet.Tablet.<init>(Tablet.java:437)
        ... 8 more
Caused by: java.io.IOException: Sort "finished" flag not found in hdfs://localhost:8020/accumulo/recovery/5bce00d4-1920-416a-8bfb-41e4cf16e9f6
        at org.apache.accumulo.tserver.log.RecoveryLogReader.<init>(RecoveryLogReader.java:138)
        at org.apache.accumulo.tserver.log.RecoveryLogsIterator.<init>(RecoveryLogsIterator.java:56)
        at org.apache.accumulo.tserver.log.SortedLogRecovery.findMaxTabletId(SortedLogRecovery.java:104)
        at org.apache.accumulo.tserver.log.SortedLogRecovery.findLogsThatDefineTablet(SortedLogRecovery.java:144)
        at org.apache.accumulo.tserver.log.SortedLogRecovery.recover(SortedLogRecovery.java:303)
        at org.apache.accumulo.tserver.log.TabletServerLogger.recover(TabletServerLogger.java:606)
</pre>
I believe the other error is being thrown in the same situation but only if the finished flag doesn't exist here:
https://github.com/apache/accumulo/blob/8d6aae599821cb43e2d320ebee5dbb480e1d41a3/server/tserver/src/main/java/org/apache/accumulo/tserver/TabletServer.java#L3303-L3312
I am not sure how the recovery can pass the first check for the finished file but then later fail, saying its not found.  Since there isn't anyting removing the file, once the finished file exists it should until the WAL is removed by the GC.

**Comment by milleruntime** [comment]
I have yet to see this error again after testing with my change in #1863.  I think I can close this once that change is merged into the main branch.

**Comment by milleruntime** [comment]
Fixed in #1863
